src/unit_test.c

```diff
@@ -0,0 +1,332 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#if defined(SUPPORTS_CONCURRENCY) && SUPPORTS_CONCURRENCY
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+#endif
+
+#include "unit_test.h"
+#include "testrand.h"
+#include "tests_common.h"
+
+#define UNUSED(x) (void)(x)
+
+/* Number of times certain tests will run */
+int COUNT = 16;
+
+static int parse_jobs_count(const char* key, const char* value, struct TestFramework* tf);
+static int parse_iterations(const char* key, const char* value, struct TestFramework* tf);
+static int parse_seed(const char* key, const char* value, struct TestFramework* tf);
+
+/* Mapping table: key → handler */
+typedef int (*ArgHandler)(const char* key, const char* value, struct TestFramework* tf);
+struct ArgMap {
+    const char* key;
+    ArgHandler handler;
+};
+
+/*
+ *   Main entry point for handling command-line arguments.
+ *
+ *   Developers should extend this map whenever new command-line
+ *   options are introduced. Each new argument should be validated,
+ *   converted to the appropriate type, and stored in 'tf->args' struct.
+ */
+static struct ArgMap arg_map[] = {
+    { "j", parse_jobs_count }, { "jobs", parse_jobs_count },
+    { "i", parse_iterations }, { "iterations", parse_iterations },
+    { "seed", parse_seed },
+    { NULL, NULL } /* sentinel */
+};
+
+/* Main entry point for reading environment variables */
+static int read_env(struct TestFramework* tf) {
+    const char* env_iter = getenv("SECP256K1_TEST_ITERS");
+    if (env_iter && strlen(env_iter) > 0) {
+        return parse_iterations("i", env_iter, tf);
+    }
+    return 0;
+}
+
+static int parse_arg(const char* key, const char* value, struct TestFramework* tf) {
+    int i;
+    for (i = 0; arg_map[i].key != NULL; i++) {
+        if (strcmp(key, arg_map[i].key) == 0) {
+            return arg_map[i].handler(key, value, tf);
+        }
+    }
+    /* Unknown key: report just so typos don’t silently pass. */
+    printf("Unknown argument '-%s=%s'\n", key, value);
+    return 0;
+}
+
+static int parse_jobs_count(const char* key, const char* value, struct TestFramework* tf) {
+    char* ptr_val;
+    long val = strtol(value, &ptr_val, 10); /* base 10 */
+    if (*ptr_val != '\0') {
+        fprintf(stderr, "Invalid number for -%s=%s\n", key, value);
+        return -1;
+    }
+    if (val < 0 || val > MAX_SUBPROCESSES) {
+        fprintf(stderr, "Arg '-%s' out of range: '%ld'. Range: 0..%d\n", key, val, MAX_SUBPROCESSES);
+        return -1;
+    }
+    tf->args.num_processes = (int) val;
+    return 0;
+}
+
+static int parse_iterations(const char* key, const char* value, struct TestFramework* tf) {
+    UNUSED(key); UNUSED(tf);
+    if (!value) return 0;
+    COUNT = (int) strtol(value, NULL, 0);
+    if (COUNT <= 0) {
+        fputs("An iteration count of 0 or less is not allowed.\n", stderr);
+        return -1;
+    }
+    printf("Iterations count = %i\n", COUNT);
```

Comment:
+1 on always printing the iteration count (also to retain the current behavior on master).